src/Core/Containers/VariableSetEnumManagement.hpp

```diff
@@ -0,0 +1,135 @@
+ * \param name The name of the enum variable
+ * \param value The value to convert
+ * \return
+ */
+template <typename Enum, typename std::enable_if<std::is_enum<Enum> {}, bool>::type = true>
+std::string getEnumString( const VariableSet& vs, const std::string& name, Enum value );
+
+/**
+ * \brief (overload) Return the string associated to the actual value of a n enumeration
+ * variable, from a value with underlying_type<Enum>. \tparam EnumBaseType The underlying enum
+ * type (\see Ra::Core::Utils::EnumConverter) \param name The name of the enum variable \param
+ * value The value to convert \return
+ */
+template <typename EnumBaseType>
+std::string
+getEnumString( const VariableSet& vs,
+               const std::string& name,
+               EnumBaseType value,
+               typename std::enable_if<!std::is_enum<EnumBaseType> {}, bool>::type = true );
+
+/**
+ * \brief set the value of the given enumeration variable, according to a string representation
+ * of an enum. \note If there is no EnumConverter associated with the variable name, the string
+ * is registered in the RenderParameter set. \param name Name of the variable \param value value
+ * of the variable
+ */
+void RA_CORE_API setEnumVariable( VariableSet& vs,
+                                  const std::string& name,
+                                  const std::string& value );
+
+void RA_CORE_API setEnumVariable( VariableSet& vs, const std::string& name, const char* value );
+
+template <typename T>
+void setEnumVariable( VariableSet& vs, const std::string& name, T value ) {
+    auto v = static_cast<typename std::underlying_type<T>::type>( value );
+    vs.setVariable( name, v );
+}
+
+template <typename T>
+auto getEnumVariable( const VariableSet& vs, const std::string& name ) -> const T& {
+    static_assert( std::is_enum<T>::value );
+    return reinterpret_cast<const T&>(
+        vs.getVariable<typename std::underlying_type<T>::type>( name ) );
+}
+
+template <typename T>
+auto getEnumVariable( VariableSet& vs, const std::string& name ) -> T& {
+    return const_cast<T&>( getEnumVariable<T>( const_cast<const VariableSet&>( vs ), name ) );
+}
+
+template <typename EnumBaseType>
+void addEnumConverter( VariableSet& vs,
+                       const std::string& name,
+                       std::shared_ptr<Core::Utils::EnumConverter<EnumBaseType>> converter ) {
+
+    // typed converter
+    auto converterHandle = vs.setVariable( name, converter );
+
+    // string string converter/setter for setEnumVariable
+    std::function<void( Core::VariableSet&, const std::string&, const std::string& )>
+        convertingFunction = [converter = converterHandle.first]( Core::VariableSet& vs,
+                                                                  const std::string& nm,
+                                                                  const std::string& vl ) {
+            vs.setVariable( nm, converter->second->getEnumerator( vl ) );
+        };
```

Comment:
This will remove a compilation warning about variable shadowing ...
```suggestion
    std::function<void( Core::VariableSet&, const std::string&, const std::string& )>
        convertingFunction = [converter = converterHandle.first]( Core::VariableSet& st,
                                                                  const std::string& nm,
                                                                  const std::string& vl ) {
            st.setVariable( nm, converter->second->getEnumerator( vl ) );
        };
```